stdlib/public/core/StringInterpolation.swift

```diff
@@ -178,7 +178,12 @@ public struct DefaultStringInterpolation: StringInterpolationProtocol {
   public mutating func appendInterpolation<T>(_ value: T) {
     _print_unlocked(value, &self)
   }
-  
+
+  @_alwaysEmitIntoClient
+  public mutating func appendInterpolation(_ value: Any.Type) {
```

Comment:
It probably makes sense to add this fast path. However:

1. I wonder if we really ought to use an existential here. If this overload instead had the signature `<T>(_ type: T.Type) -> Void`, would it be easier for the optimizer (possibly in the future if not now) to potentially evaluate this at compile time and produce a string literal?

2. Should we also add equivalent fast paths for `String.init(describing:)` and `String.init(reflecting:)`?

Reply:
1. Since this is emit-into-client, it doesn't really matter which might be better to constant fold in the future, since we can just switch to the other technique if necessary. For now, my inclination is to prefer a non-generic to generic method if there's no difference to the actual code generated.

2. Maybe. I'm slightly more wary about adding overloads to `String` than I am to `DefaultStringInterpolation` though. Maybe without basis.

Reply:
From @ravikandhadai 
> At present the constant evaluator works better with generics rather than existentials. But this is only an implementation-level restriction. Ideally, I think as long as it is passed a static constant (like Type.self) it can be folded, regardless of using generic or existential.